docs/src/examples/custom-relu.jl

```diff
@@ -64,9 +64,9 @@ function ChainRulesCore.rrule(
             obj_exp = MOI.get(
                 model,
                 DiffOpt.BackwardOutObjective()
-            ) # return grdiente wrt objective function parameters
-            dy[:, i] = JuMP.coefficient.(obj_exp, x) # coeff of `x` in -2x'y
-            dy[:, i] = -2 * dy[:, i]
+            ) # return gradient wrt objective function parameters
+            dq[:, i] = JuMP.coefficient.(obj_exp, x) # coeff of `x` in q'x = -2x'y
+            dy[:, i] = -2 * dq[:, i]  # ∵ dq/dy = -2
```

Comment:
If `dq/dy = -2` then it should be `dy[:, i] = -dq[:, i] / 2`.

Reply:
good catch

Reply:
```suggestion
            dy[:, i] = -2dq[:, i]/2  # q = -2y, so dq/dy = -2 and dy = -dq/2
```

Reply:
Nooooooo, that's why I was really confused with this package.
As I pointed out in [a comment, #178](https://github.com/jump-dev/DiffOpt.jl/issues/176#issuecomment-989469397), the `dy` actually refers to `dl/dy` for some output `l` (e.g., loss).

That is, the input of the pullback, `dx`, means `dl/dx`.
So `dy` and `dq` correspond to `dl/dy` and `dl/dq`, respectively.
Therefore, to convert `dq` to `dy`, we need this calculation:
`dl/dy = dl/dq * dq/dy`,
which is notationally opposite to the mathematical expression of `dy = -2 * dq` (sooo confusing).

Reply:
For example, in thie ReLU example, we will get `dl/dy`, denoted by `dy` in the code.
When letting `dx` (actually means `dl/dx`) be 1, we will get
`dl/dy = dl/dx * dx/dy = dx/dy`.

Since the activation ReLU provides a subgradient of
\in {0}, if `y < 0`
\in [0, 1], if `y = 0`
\in {1}, if `y > 0`
so whether the output is correct can be checked by applying any `y` into as
```julia
    m = 1
    d = 1
    y = ones(m, d)
    @show matrix_relu(y)
    pv, pullback_matrix_relu = _rrule(y)  # corresponds to ChainRulesCore.rrule
    dx = ones(m, d)  # given dl/dx
    pullback_matrix_relu(dx)  # dl/dy = dl/dx * dx/dy = dx/dy
```
gives us
```julia
dy[:, i] = -2 * dq[:, i] = [1.0]
```
which is correct.

Reply:
To make it clear, I added it to docs.

Reply:
might well be, i'd have to think more carefully.
if dy is actually `dl/dy,` we have to rename it to `dl_dy`.
same for `dq`.

Reply:
same for `dx`

Reply:
It would be much better to use some different symbols (for example, \dot for forward mode and \overline for reverse mode, e.g., \dot{x}) as [ChainRulesCore.jl](https://juliadiff.org/ChainRulesCore.jl/dev/#Reverse-mode-AD-rules-(rrules)) to distinguish them from `dfoo`.

To avoid some Unicode issues, how 'bout `_x` for forward mode and `x_` for reverse mode?
That is, `_x = dx/da` for some `a`, and `x_ = dl/dx` for some `l`.

Reply:
I dont see why `x_` is better than `dl_dx` to represent `dl/dx`. The first requires that the user reads a manual saying `x_ = dl/dx`, while the second is much close to what is meant.

Reply:
You're right, it's clearer. Honestly, both of them are fine to me :)
But in that case, users should know what `l` refers to as well.
So we would have to put some additional comments either which one is taken.

Reply:
I agree with the need for additional comments in both options.